contracts/verifiers/Verifier.sol

```diff
@@ -421,8 +428,12 @@ abstract contract Verifier is IVerifier, ContextUpgradeable {
         address sender,
         string memory responseFieldName
     ) public view checkRequestExistence(requestId, true) returns (uint256) {
+        if (!isRequestProofVerified(sender, requestId)) {
+            revert ProofIsNotVerified(requestId, sender);
+        }
         VerifierStorage storage s = _getVerifierStorage();
-        return s._proofs[requestId][sender].responseFields[responseFieldName];
+        Proof storage proof = s._proofs[requestId][sender];
+        return proof.proofEntries[proof.proofEntries.length - 1].responseFields[responseFieldName];
```

Comment:
Here you have to check if the proof is verified because it could have no entry.
```suggestion
        if (!proof.isVerified) {
           revert ProofIsNotVerified(requestId, sender); 
        }
        return proof.proofEntries[proof.proofEntries.length - 1].responseFields[responseFieldName];
```

Reply:
Fixed